backend.native/tests/runtime/collections/array5.kt

```diff
@@ -0,0 +1,814 @@
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[Int.MAX_VALUE] = 1.0
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[Int.MIN_VALUE] = 1.0
+    }
+}
+
+@Test fun booleanArrayGet() {
+    val arr = BooleanArray(10) { false }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[10]
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[-1]
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[Int.MAX_VALUE]
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[Int.MIN_VALUE]
+    }
+}
+
+@Test fun booleanArraySet() {
+    val arr = BooleanArray(10) { false }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[10] = true
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[-1] = true
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[Int.MAX_VALUE] = true
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[Int.MIN_VALUE] = true
+    }
+}
+
+@Test fun charArrayGet() {
+    val arr = CharArray(10) { '0' }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[10]
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[-1]
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[Int.MAX_VALUE]
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[Int.MIN_VALUE]
+    }
+}
+
+@Test fun charArraySet() {
+    val arr = CharArray(10) { '0' }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[10] = '1'
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[-1] = '1'
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[Int.MAX_VALUE] = '1'
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr[Int.MIN_VALUE] = '1'
+    }
+}
+
+@Test fun byteArrayGetUByte() {
+    val arr = ByteArray(10) { 0 }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr.getUByteAt(10)
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr.getUByteAt(-1)
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr.getUByteAt(Int.MAX_VALUE)
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr.getUByteAt(Int.MIN_VALUE)
+    }
+}
+
+@Test fun byteArraySetUByte() {
+    val arr = ByteArray(10) { 0 }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr.setUByteAt(10, 1U)
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr.setUByteAt(-1, 1U)
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr.setUByteAt(Int.MAX_VALUE, 1U)
+    }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr.setUByteAt(Int.MIN_VALUE, 1U)
+    }
+}
+
+@Test fun byteArrayGetChar() {
+    val arr = ByteArray(10) { 0 }
+    assertFailsWith<ArrayIndexOutOfBoundsException> {
+        arr.getCharAt(10)
```

Comment:
Note that `10` is not a minimal positive index causing `ArrayIndexOutOfBoundsException`. `9` should too.
Same below.

Reply:
Thanks, added some positive cases as well.